src/Installer/finalizer/Program.cs

```diff
@@ -0,0 +1,246 @@
+      return false;
+    }
+
+    foreach (var valueName in key.GetValueNames())
+    {
+      if (valueName.Contains(featureBandVersion))
+      {
+        Console.WriteLine($"SDK version detected: {valueName}");
+        return true;
+      }
+    }
+  }
+  return false;
+}
+
+static bool RemoveDependent(string dependent)
+{
+    Installer.SetInternalUI(InstallUIOptions.Silent);
+
+    // Open the installer dependencies registry key
+    // This has to be an exhaustive search as we're not looking for a specific provider key, but for a specific dependent
+    // that could be registered against any provider key.
+    using var hkInstallerDependenciesKey = Registry.LocalMachine.OpenSubKey(@"SOFTWARE\Classes\Installer\Dependencies", writable: true);
+    if (hkInstallerDependenciesKey == null)
+    {
+        Console.WriteLine("Installer dependencies key does not exist.");
+        return false; // No dependencies to remove
+    }
+
+    // Iterate over each provider key in the dependencies
+    foreach (string providerKeyName in hkInstallerDependenciesKey.GetSubKeyNames())
+    {
+        Console.WriteLine($"Processing provider key: {providerKeyName}");
+
+        using var hkProviderKey = hkInstallerDependenciesKey.OpenSubKey(providerKeyName, writable: true);
+        if (hkProviderKey == null) continue;
+
+        // Open the Dependents subkey
+        using var hkDependentsKey = hkProviderKey.OpenSubKey("Dependents", writable: true);
+        if (hkDependentsKey == null) continue;
+
+        // Check if the dependent exists and continue if it does not
+        string[] dependentsKeys = hkDependentsKey.GetSubKeyNames();
+        bool dependentExists = false;
+
+        foreach (string dependentsKeyName in dependentsKeys)
+        {
+            if (string.Equals(dependentsKeyName, dependent, StringComparison.OrdinalIgnoreCase))
+            {
+                dependentExists = true;
+                break;
+            }
+        }
+
+        if (!dependentExists)
+        {
+            continue; // Skip to the next provider key if the dependent does not exist
+        }
+
+        Console.WriteLine($"Dependent match found: {dependent}");
+
+        // Attempt to remove the dependent key
+        try
+        {
+            hkDependentsKey.DeleteSubKey(dependent);
+            Console.WriteLine("Dependent deleted");
+        }
+        catch (Exception ex)
+        {
+            Console.WriteLine($"Exception while removing dependent key: {ex.Message}");
+            return false;
+        }
+
+        // Check if any dependents are left
+        if (hkDependentsKey.SubKeyCount == 0)
+        {
+            // No remaining dependents, handle product uninstallation
+            try
+            {
+                string productCode = hkProviderKey.GetValue("ProductId").ToString();
+
+                // Configure the product to be absent
+                Installer.ConfigureProduct(productCode, 0, InstallState.Absent, "");
```

Comment:
We can remove the DTF dependency by just invoking the API directly. Try adding the cswin32 package to the .csproj. We already use cswin32 in the dotnet CLI

1. Add a package reference to cswin32 - this is a code generator for native APIs that generated based off Window metadata package. Something like `<PackageReference Include="Microsoft.Windows.CsWin32" />`
2. Add a file called `NativeMethods.txt` next to the .csproj and include the following text
```
ERROR_*
Msi.*
```
3. You should be able to then add 

```
using Windows.Win32;
using Windows.Win32.Foundation;
using Windows.Win32.System.ApplicationInstallationAndServicing;
```
and then replace `Installer.ConfigureProduct` call with
```csharp
unit error = PInvoke.ConfigureProductEx(productCode, INSTALLLEVEL.INSTALLLEVEL_DEFAULT, INSTALLSTATE.INSTALLSTATE_ABSENT, "");
```